invenio_rdm_records/resources/resources.py

```diff
@@ -368,6 +369,24 @@ def get_suggestions(self):
         )
         return items.to_dict(), 200
 
+    @request_view_args
+    @request_data
+    def set_default(self):
+        """Set default community."""
+        item, errors = self.service.set_default(
+            id_=resource_requestctx.view_args["pid_value"],
+            identity=g.identity,
+            data=resource_requestctx.data,
+        )
+
+        response = {}
+        if item:
+            response["result"] = item
+        if errors:
+            response["errors"] = errors
```

Comment:
Major: We usually manipulate the response and fill it with errors when we allow to pass multiple values that will be processed and some will success and others might fail. It doesn't seem the case here, we can only set one default community, therefore I would suggest to use error handlers to handle the error. You will need to create a error for [here](https://github.com/inveniosoftware/invenio-communities/pull/1033/files#diff-e062fa3bca790c1a96331a1093f0b6c354d7364099ad90915f96f72d9f046236R170) and then you add the handler for that error [here](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/resources/config.py#L401).